payments/core.py

```diff
@@ -156,6 +161,12 @@ def provider_factory(variant: str):
     return PROVIDER_CACHE[variant]
 
 
+PAYMENT_VARIANT_FACTORY = getattr(settings, "PAYMENT_VARIANT_FACTORY", None)
+if PAYMENT_VARIANT_FACTORY:
+    provider_factory = import_string(PAYMENT_VARIANT_FACTORY)
+else:
+    provider_factory = _default_provider_factory
+
```

Comment:
Hmm, I am not convinced. Defining a name, then importing that name from the current module again, then reassign the same thing to the same name again... That does not strike me as readable. It might technically work, but having the import_string try to import the same module during its own import time seems awkward.

Reply:
I think I initially misinterpreted how this works, you're still exposing `provider_factory`, but based on the current setting. :+1: 

I think this makes more sense (rather than special-case the unconfigured scenario, just provide a default).
```py
PAYMENT_VARIANT_FACTORY = getattr(settings, "PAYMENT_VARIANT_FACTORY", "payments.core._default_provider_factory")
provider_factory = import_string(PAYMENT_VARIANT_FACTORY)
```

Reply:
This still implies that at this point in the code, you are calling Python to import a name from the same module you are making the import in (you are doing `from payments.core import ...` in the `payments.core` module). I am not sure this is guaranteed to work properly!